src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -247,52 +241,60 @@ osexperimental_handle_download_packages (RPMOSTreeOSExperimental *interface,
           hy_query_filter (query, HY_PKG_REPONAME, HY_EQ, source_name);
           pkglist = hy_query_run (query);
           if (!pkglist || pkglist->len == 0)
-            {
-              glnx_throw (&local_error, "No matches for \"%s\" in repo '%s'", pkg_name, source_name);
-              goto out;
-            }
+            return glnx_prefix_error (error, "No matches for \"%s\" in repo '%s'", pkg_name, source_name);
+
           g_ptr_array_add (dnf_pkgs, g_object_ref (pkglist->pdata[0]));
         }
     }
   /* Future source kinds go here */
   else
-    {
-       glnx_throw (&local_error, "Unsupported source type: %s", source);
-       goto out;
-    }
+    return glnx_prefix_error (error, "Unsupported source type: %s", source);
 
   rpmostree_set_repos_on_packages (rpmostree_context_get_dnf (ctx), dnf_pkgs);
 
-  if (!rpmostree_download_packages (dnf_pkgs, cancellable, &local_error))
-    goto out;
+  if (!rpmostree_download_packages (dnf_pkgs, cancellable, error))
+    return glnx_prefix_error (error, "Downloading packages");
 
-  out_fd_list = g_unix_fd_list_new ();
+  fd_list = g_unix_fd_list_new ();
   for (unsigned int i = 0;  i < dnf_pkgs->len; i++)
     {
       DnfPackage *pkg = static_cast<DnfPackage *>(dnf_pkgs->pdata[i]);
       const gchar *path = dnf_package_get_filename (pkg);
       gint fd = -1;
 
-      if (!glnx_openat_rdonly (AT_FDCWD, path, TRUE, &fd, &local_error))
-        goto out;
+      if (!glnx_openat_rdonly (AT_FDCWD, path, TRUE, &fd, error))
+        return FALSE;
 
-      if (g_unix_fd_list_append (out_fd_list, fd, &local_error) < 0)
-        goto out; 
+      if (g_unix_fd_list_append (fd_list, fd, error) < 0)
+        return FALSE;
 
-      if (!glnx_unlinkat (AT_FDCWD, path, 0, &local_error))
-        goto out;
+      if (!glnx_unlinkat (AT_FDCWD, path, 0, error))
+        return FALSE;
     }
 
-out:
+  *out_fd_list = util::move_nullify (fd_list);
+  return TRUE;
+}
+
+static gboolean
+osexperimental_handle_download_packages (RPMOSTreeOSExperimental *interface,
+                                         GDBusMethodInvocation *invocation,
+                                         GUnixFDList* _fds,
```

Comment:
This parameter seemed unused in the original code, but I think it's just the result of dbus funky way of handling "out" parameters.